diff --git config/adv360.keymap config/adv360.keymap
index cf80f15..e4be970 100644
--- config/adv360.keymap
+++ config/adv360.keymap
@@ -1,71 +1,93 @@
+
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+
+/* THIS FILE WAS GENERATED!
+ *
+ * This file was generated automatically. You may or may not want to
+ * edit it directly.
+ */
+
 #include <behaviors.dtsi>
 #include <dt-bindings/zmk/keys.h>
 #include <dt-bindings/zmk/bt.h>
+#include <dt-bindings/zmk/outputs.h>
 #include <dt-bindings/zmk/rgb.h>
 #include <dt-bindings/zmk/backlight.h>
 
 / {
-    behaviors {
-      #include "macros.dtsi"
-      #include "version.dtsi"
-      #ifndef VERSION_MACRO
-      macro_ver: macro_ver {
-        compatible = "zmk,behavior-macro";
-        label = "macro_version";
-        #binding-cells = <0>;
-        bindings = <&kp RET>;
-      };
-      #endif
+  behaviors {
+    #include "macros.dtsi"
+    #include "version.dtsi"
 
-      hm: homerow_mods {
-          compatible = "zmk,behavior-hold-tap";
-          label = "HOMEROW_MODS";
-          #binding-cells = <2>;
-          tapping-term-ms = <200>;
-          quick_tap_ms = <175>;
-          flavor = "tap-preferred";
-          bindings = <&kp>, <&kp>;
-      };
+    hm: homerow_mods {
+        compatible = "zmk,behavior-hold-tap";
+        label = "HOMEROW_MODS";
+        #binding-cells = <2>;
+        tapping-term-ms = <200>;
+        quick_tap_ms = <175>;
+        flavor = "tap-preferred";
+        bindings = <&kp>, <&kp>;
     };
+  };
 
-  keymap {
-    compatible = "zmk,keymap";
+    keymap {
+        compatible = "zmk,keymap";
+
+
+        default_layer {
+            bindings = <
+ &kp EQUAL               &kp N1           &kp N2    &kp N3     &kp N4  &kp N5               &tog 1                                                                         &mo 3  &kp N6  &kp N7     &kp N8    &kp N9    &kp N0  &kp MINUS
+   &kp TAB                &kp Q            &kp W     &kp E      &kp R   &kp T                &none                                                              &kp LEFT_BRACKET   &kp Y   &kp U      &kp I     &kp O     &kp P   &kp BSLH
+   &kp ESC                &kp A            &kp S     &kp D      &kp F   &kp G            &kp GRAVE  &kp LCTRL  &kp LALT                   &kp LGUI  &kp RCTRL  &kp RIGHT_BRACKET   &kp H   &kp J      &kp K     &kp L  &kp SEMI    &kp SQT
+ &kp LSHFT                &kp Z            &kp X     &kp C      &kp V   &kp B                                  &kp HOME                  &kp PG_UP                                 &kp N   &kp M  &kp COMMA   &kp DOT  &kp FSLH  &kp RSHFT
+     &mo 2  &macro_mac_profile1  &kp PRINTSCREEN  &kp LEFT  &kp RIGHT                     &kp BSPC    &kp DEL   &kp END                  &kp PG_DN  &kp ENTER          &kp SPACE          &kp UP   &kp DOWN  &kp LBKT  &kp RBKT      &mo 2
+            >;
+        };
+
+        layer_keypad {
+            bindings = <
+ &kp EQUAL  &kp N1  &kp N2    &kp N3     &kp N4  &kp N5              &trans                                                                 &mo 3  &kp N6  &kp KP_NUM  &kp KP_EQUAL  &kp KP_DIVIDE  &kp KP_MULTIPLY  &kp MINUS
+   &kp TAB   &kp Q   &kp W     &kp E      &kp R   &kp T               &none                                                                 &none   &kp Y   &kp KP_N7     &kp KP_N8      &kp KP_N9     &kp KP_MINUS   &kp BSLH
+   &kp ESC   &kp A   &kp S     &kp D      &kp F   &kp G               &none  &kp LCTRL  &kp LALT                   &kp LGUI  &kp RCTRL      &none   &kp H   &kp KP_N4     &kp KP_N5      &kp KP_N6      &kp KP_PLUS    &kp SQT
+ &kp LSHFT   &kp Z   &kp X     &kp C      &kp V   &kp B                                 &kp HOME                  &kp PG_UP                         &kp N   &kp KP_N1     &kp KP_N2      &kp KP_N3     &kp KP_ENTER  &kp RSHFT
+    &trans  &trans  &trans  &kp LEFT  &kp RIGHT                    &kp BSPC    &kp DEL   &kp END                  &kp PG_DN  &kp ENTER  &kp KP_N0              &kp UP      &kp DOWN     &kp KP_DOT         &kp RBKT     &trans
+            >;
+        };
+
+        layer_fn {
+            bindings = <
+ &kp F1  &kp F2  &kp F3  &kp F4  &kp F5  &kp F6            &tog 1                                                   &mo 3  &kp F7  &kp F8  &kp F9  &kp F10  &kp F11  &kp F12
+ &trans  &trans  &trans  &trans  &trans  &trans             &none                                                   &none  &trans  &trans  &trans   &trans   &trans   &trans
+ &trans  &trans  &trans  &trans  &trans  &trans             &none  &trans  &trans                  &trans  &trans   &none  &trans  &trans  &trans   &trans   &trans   &trans
+ &trans  &trans  &trans  &trans  &trans  &trans                            &trans                  &trans                  &trans  &trans  &trans   &trans   &trans   &trans
+ &trans  &trans  &trans  &trans  &trans                    &trans  &trans  &trans                  &trans  &trans  &trans          &trans  &trans   &trans   &trans   &trans
+            >;
+        };
+
+        layer_mod {
+            bindings = <
+  &none         &bt BT_SEL 0         &bt BT_SEL 1  &bt BT_SEL 2  &bt BT_SEL 3  &bt BT_SEL 4                  &none                                                                          &trans   &none  &out OUT_USB  &out OUT_BLE   &none   &none                  &none
+  &none                &none                &none         &none         &none         &none            &bootloader                                                                     &bootloader   &none         &none         &none   &none   &none                  &none
+  &none                &none                &none         &none         &none         &none                  &none   &none   &none                  &bt BT_CLR       &none  &rgb_ug RGB_MEFS_CMD 5   &none         &none         &none   &none   &none                  &none
+  &none                &none                &none         &none         &none         &none                                  &none                       &none                                       &none         &none         &none   &none   &none                  &none
+  &none  &macro_mac_profile1  &macro_mac_profile3         &none         &none                                &none   &none   &none                       &none  &bl BL_TOG         &rgb_ug RGB_TOG            &bl BL_INC    &bl BL_DEC   &none   &none  &macro_linux_profile0
+            >;
+        };
+
+        layer_MAC {
+            bindings = <
+ &trans  &trans  &trans  &trans  &trans  &trans            &trans                                                                        &mo 3  &trans  &trans  &trans  &trans  &trans                 &trans
+ &trans  &trans  &trans  &trans  &trans  &trans            &trans                                                                       &trans  &trans  &trans  &trans  &trans  &trans                 &trans
+ &trans  &trans  &trans  &trans  &trans  &trans            &trans  &kp LEFT_COMMAND  &kp LEFT_ALT                  &kp LGUI  &kp RCTRL  &trans  &trans  &trans  &trans  &trans  &trans                 &trans
+ &trans  &trans  &trans  &trans  &trans  &trans                                            &trans                    &trans                     &trans  &trans  &trans  &trans  &trans                 &trans
+ &trans  &trans  &trans  &trans  &trans                    &trans            &trans        &trans                    &trans     &trans  &trans          &trans  &trans  &trans  &trans  &macro_linux_profile0
+            >;
+        };
 
-    default_layer {
-      bindings = <
-        &kp EQUAL &kp N1    &kp N2   &kp N3   &kp N4     &kp N5 &tog 1                                                           &mo 3 &kp N6 &kp N7 &kp N8    &kp N9   &kp N0   &kp MINUS
-        &kp TAB   &kp Q     &kp W    &kp E    &kp R      &kp T  &none                                                            &none &kp Y  &kp U  &kp I     &kp O    &kp P    &kp BSLH
-        &kp ESC   &kp A     &kp S    &kp D    &kp F      &kp G  &none           &kp LCTRL &kp LALT &kp LGUI  &kp RCTRL           &none &kp H  &kp J  &kp K     &kp L    &kp SEMI &kp SQT
-        &kp LSHFT &kp Z     &kp X    &kp C    &kp V      &kp B                            &kp HOME &kp PG_UP                           &kp N  &kp M  &kp COMMA &kp DOT  &kp FSLH &kp RSHFT
-        &mo 2     &kp GRAVE &kp CAPS &kp LEFT &kp RIGHT                &kp BSPC &kp DEL   &kp END  &kp PG_DN &kp ENTER &kp SPACE              &kp UP &kp DOWN  &kp LBKT &kp RBKT &mo 2
-      >;
-    };
-    keypad {
-      bindings = <
-        &kp EQUAL &kp N1    &kp N2   &kp N3   &kp N4     &kp N5 &trans                                                           &mo 3 &kp N6 &kp KP_NUM &kp KP_EQUAL &kp KP_DIVIDE &kp KP_MULTIPLY &kp MINUS
-        &kp TAB   &kp Q     &kp W    &kp E    &kp R      &kp T  &none                                                            &none &kp Y  &kp KP_N7  &kp KP_N8    &kp KP_N9     &kp KP_MINUS    &kp BSLH
-        &kp ESC   &kp A     &kp S    &kp D    &kp F      &kp G  &none           &kp LCTRL &kp LALT &kp LGUI  &kp RCTRL           &none &kp H  &kp KP_N4  &kp KP_N5    &kp KP_N6     &kp KP_PLUS     &kp SQT
-        &kp LSHFT &kp Z     &kp X    &kp C    &kp V      &kp B                            &kp HOME &kp PG_UP                           &kp N  &kp KP_N1  &kp KP_N2    &kp KP_N3     &kp KP_ENTER    &kp RSHFT
-        &mo 2     &kp GRAVE &kp CAPS &kp LEFT &kp RIGHT                &kp BSPC &kp DEL   &kp END  &kp PG_DN &kp ENTER &kp KP_N0              &kp UP     &kp DOWN     &kp KP_DOT    &kp RBKT        &mo 2
-      >;
-    };
-    fn {
-      bindings = <
-        &kp F1 &kp F2 &kp F3 &kp F4 &kp F5 &kp F6 &tog 1                                           &mo 3 &kp F7 &kp F8 &kp F9 &kp F10 &kp F11 &kp F12
-        &trans &trans &trans &trans &trans &trans  &none                                           &none &trans &trans &trans &trans  &trans  &trans
-        &trans &trans &trans &trans &trans &trans  &none        &trans &trans &trans &trans        &none &trans &trans &trans &trans  &trans  &trans
-        &trans &trans &trans &trans &trans &trans                      &trans &trans                     &trans &trans &trans &trans  &trans  &trans
-        &trans &trans &trans &trans &trans               &trans &trans &trans &trans &trans &trans              &trans &trans &trans  &trans  &trans
-      >;
-    };
-    mod {
-      bindings = <
-        &none &bt BT_SEL 0 &bt BT_SEL 1 &bt BT_SEL 2 &bt BT_SEL 3 &bt BT_SEL 4 &none                                                                     &trans                 &none        &none        &none        &none &none &none
-        &none &none        &none        &none        &none        &none        &bootloader                                                               &bootloader            &none        &none        &none        &none &none &none
-        &none &none        &none        &none        &none        &none        &none                   &none &none &bt BT_CLR &none                      &rgb_ug RGB_MEFS_CMD 5 &none        &none        &none        &none &none &none
-        &none &none        &none        &none        &macro_ver   &none                                      &none &none                                                        &none        &none        &none        &none &none &none
-        &none &none        &none        &none        &none                                       &none &none &none &none      &bl BL_TOG &rgb_ug RGB_TOG                                     &bl BL_INC   &bl BL_DEC   &none &none &none
-      >;
     };
-  };
 };
diff --git config/boards/arm/adv360/Kconfig.defconfig config/boards/arm/adv360/Kconfig.defconfig
index 7333f89..aa2fb87 100644
--- config/boards/arm/adv360/Kconfig.defconfig
+++ config/boards/arm/adv360/Kconfig.defconfig
@@ -35,6 +35,12 @@ config SPI
 config BT_CTLR
     default BT
 
+config ZMK_BLE
+    default y
+
+config ZMK_USB
+    default y
+
 if USB
 
 config USB_NRFX
diff --git config/boards/arm/adv360/adv360.dtsi config/boards/arm/adv360/adv360.dtsi
index 0c85836..2f221fc 100644
--- config/boards/arm/adv360/adv360.dtsi
+++ config/boards/arm/adv360/adv360.dtsi
@@ -21,7 +21,6 @@
         zephyr,code-partition = &code_partition;
         zephyr,sram = &sram0;
         zephyr,flash = &flash0;
-        zephyr,console = &cdc_acm_uart;
         zmk,kscan = &kscan0;
         zmk,backlight = &backlight;
         zmk,battery = &vbatt;
@@ -44,8 +43,9 @@
 
         >;
     };
-    EXT_POWER {
+    ext-power {
             compatible = "zmk,ext-power-generic";
+            label = "EXT_POWER";
             control-gpios = <&gpio0 13 GPIO_ACTIVE_HIGH>;
         };
 
@@ -90,9 +90,6 @@
 
 &usbd {
     status = "okay";
-    cdc_acm_uart: cdc_acm_uart {
-        compatible = "zephyr,cdc-acm-uart";
-    };
 };
 
 
@@ -107,9 +104,11 @@
         #size-cells = <1>;
 
         sd_partition: partition@0 {
+            label = "softdevice";
             reg = <0x00000000 0x00026000>;
         };
         code_partition: partition@26000 {
+            label = "code_partition";
             reg = <0x00026000 0x000c6000>;
         };
 
@@ -123,10 +122,12 @@
          * if enabled.
          */
         storage_partition: partition@ec000 {
+            label = "storage";
             reg = <0x000ec000 0x00008000>;
         };
 
         boot_partition: partition@f4000 {
+            label = "adafruit_boot";
             reg = <0x000f4000 0x0000c000>;
         };
     };
@@ -141,6 +142,7 @@
 
     led_strip: ws2812@0 {
         compatible = "worldsemi,ws2812-spi";
+        label = "WS2812";
 
         /* SPI */
         reg = <0>;
diff --git config/boards/arm/adv360/adv360_left_defconfig config/boards/arm/adv360/adv360_left_defconfig
index b37679a..0b88cfb 100644
--- config/boards/arm/adv360/adv360_left_defconfig
+++ config/boards/arm/adv360/adv360_left_defconfig
@@ -49,26 +49,21 @@ CONFIG_ZMK_BACKLIGHT_AUTO_OFF_IDLE=y
 #BT configuration
 CONFIG_BT_CTLR_TX_PWR_PLUS_8=y
 CONFIG_BT_DIS_MANUF="Kinesis Corporation"
-#CONFIG_BT_PRIVACY=n
-#CONFIG_BT_GAP_PERIPHERAL_PREF_PARAMS=y
-#CONFIG_BT_PERIPHERAL_PREF_TIMEOUT=600
-#CONFIG_BT_PERIPHERAL_PREF_MIN_INT=12
-#CONFIG_BT_PERIPHERAL_PREF_MAX_INT=24
-#CONFIG_BT_PERIPHERAL_PREF_LATENCY=30
-CONFIG_BT_BAS=n #Set this to y to enable BLE battery reporting
+CONFIG_BT_PRIVACY=y
+CONFIG_BT_GAP_PERIPHERAL_PREF_PARAMS=y
+CONFIG_BT_PERIPHERAL_PREF_TIMEOUT=600
+CONFIG_BT_PERIPHERAL_PREF_MIN_INT=12
+CONFIG_BT_PERIPHERAL_PREF_MAX_INT=24
+CONFIG_BT_PERIPHERAL_PREF_LATENCY=30
+# Battery reporting
+# https://zmk.dev/docs/config/system#bluetooth
+# https://github.com/KinesisCorporation/Adv360-Pro-ZMK/pull/248/files
+CONFIG_BT_BAS=y
 
 #Misc configuration
 CONFIG_USB_DEVICE_VID=0x29EA
 CONFIG_USB_DEVICE_PID=0x0362
 CONFIG_USB_DEVICE_MANUFACTURER="Kinesis Corporation"
 CONFIG_ZMK_HID_REPORT_TYPE_NKRO=y
-CONFIG_ZMK_HID_KEYBOARD_EXTENDED_REPORT=n #Set this to y for F13-F24 and intl keys with NKRO
 CONFIG_ZMK_HID_CONSUMER_REPORT_USAGES_BASIC=y
 CONFIG_BUILD_OUTPUT_UF2=y
-
-CONFIG_ZMK_BLE_EXPERIMENTAL_FEATURES=y
-CONFIG_ZMK_BLE_PASSKEY_ENTRY=n
-
-CONFIG_ZMK_BLE=y
-CONFIG_ZMK_USB=y
-CONFIG_ZMK_HID_INDICATORS=y
\ No newline at end of file
diff --git config/boards/arm/adv360/adv360_right_defconfig config/boards/arm/adv360/adv360_right_defconfig
index 3c2e65a..f1eb05f 100644
--- config/boards/arm/adv360/adv360_right_defconfig
+++ config/boards/arm/adv360/adv360_right_defconfig
@@ -54,8 +54,3 @@ CONFIG_BT_CTLR_TX_PWR_PLUS_8=y
 CONFIG_ZMK_HID_REPORT_TYPE_NKRO=y
 CONFIG_ZMK_HID_CONSUMER_REPORT_USAGES_BASIC=y
 CONFIG_BUILD_OUTPUT_UF2=y
-
-CONFIG_ZMK_BLE_EXPERIMENTAL_FEATURES=y
-
-CONFIG_ZMK_BLE=y
-CONFIG_ZMK_USB=y
diff --git config/boards/arm/adv360/board.cmake config/boards/arm/adv360/board.cmake
index 292a274..3465290 100644
--- config/boards/arm/adv360/board.cmake
+++ config/boards/arm/adv360/board.cmake
@@ -5,5 +5,5 @@
 
 board_runner_args(nrfjprog "--nrf-family=NRF52" "--softreset")
 
-include(${ZEPHYR_BASE}/boards/common/uf2.board.cmake)
 include(${ZEPHYR_BASE}/boards/common/nrfjprog.board.cmake)
+#include(${ZEPHYR_BASE}/boards/common/blackmagicprobe.board.cmake)
diff --git config/cust_behaviors.json config/cust_behaviors.json
new file mode 100644
index 0000000..ba42f24
--- /dev/null
+++ config/cust_behaviors.json
@@ -0,0 +1,10 @@
+[
+	{
+		"code": "RIGHT_COMMAND",
+		"name": "RIGHT_COMMAND",
+		"params": [
+			"code"
+		],
+		"type": "UD"
+	}
+]
\ No newline at end of file
diff --git config/cust_keycodes.json config/cust_keycodes.json
new file mode 100644
index 0000000..f17b6ef
--- /dev/null
+++ config/cust_keycodes.json
@@ -0,0 +1,19 @@
+[
+	{
+		"names": [
+			"&kp RIGHT_COMMAND"
+		],
+		"description": "&kp RIGHT_COMMAND",
+		"clarify": false,
+		"documentation": "",
+		"os": {
+			"windows": true,
+			"linux": true,
+			"android": true,
+			"macos": true,
+			"ios": true
+		},
+		"footnotes": {},
+		"type": "UD"
+	}
+]
\ No newline at end of file
diff --git config/keymap.json config/keymap.json
index bd3fa91..3e69cd6 100644
--- config/keymap.json
+++ config/keymap.json
@@ -6,33 +6,40 @@
     "base",
     "keypad",
     "fn",
-    "mod"
+    "mod",
+    "MAC"
   ],
   "layers": [
     [
-      "&kp EQUAL ", "&kp N1    ", "&kp N2   ", "&kp N3   ", "&kp N4     ", "&kp N5 ", "&tog 1                                                                            ", "&mo 3  ", "&kp N6 ", "&kp N7 ", "&kp N8    ", "&kp N9   ", "&kp N0   ", "&kp MINUS",
-      "&kp TAB   ", "&kp Q     ", "&kp W    ", "&kp E    ", "&kp R      ", "&kp T  ", "&none                                                                             ", "&none  ", "&kp Y  ", "&kp U  ", "&kp I     ", "&kp O    ", "&kp P    ", "&kp BSLH",
-      "&kp ESC   ", "&kp A     ", "&kp S    ", "&kp D    ", "&kp F      ", "&kp G  ", "&none           ", "&kp LCTRL ", "&kp LALT ", "&kp LGUI  ", "&kp RCTRL",             "&none  ", "&kp H  ", "&kp J  ", "&kp K     ", "&kp L    ", "&kp SEMI ", "&kp SQT",
-      "&kp LSHFT ", "&kp Z     ", "&kp X    ", "&kp C    ", "&kp V      ", "&kp B   ",                                  "&kp HOME ", "&kp PG_UP ",                          "&kp N  ", "&kp M  ", "&kp COMMA ", "&kp DOT  ", "&kp FSLH ", "&kp RSHFT",
-      "&mo 2     ", "&kp GRAVE ", "&kp CAPS ", "&kp LEFT ", "&kp RIGHT                   ", "&kp BSPC  ", "&kp DEL   ", "&kp END  ", "&kp PG_DN ", "&kp ENTER ", "&kp SPACE                    ", "&kp UP ", "&kp DOWN  ", "&kp LBKT ", "&kp RBKT ", "&mo 2"
+       "&kp EQUAL",              "&kp N1",          "&kp N2",   "&kp N3",    "&kp N4", "&kp N5",              "&tog 1",                                                                                    "&mo 3", "&kp N6", "&kp N7",    "&kp N8",   "&kp N9",   "&kp N0", "&kp MINUS",
+         "&kp TAB",               "&kp Q",           "&kp W",    "&kp E",     "&kp R",  "&kp T",               "&none",                                                                         "&kp LEFT_BRACKET",  "&kp Y",  "&kp U",     "&kp I",    "&kp O",    "&kp P",  "&kp BSLH",
+         "&kp ESC",               "&kp A",           "&kp S",    "&kp D",     "&kp F",  "&kp G",           "&kp GRAVE", "&kp LCTRL", "&kp LALT",                      "&kp LGUI", "&kp RCTRL", "&kp RIGHT_BRACKET",  "&kp H",  "&kp J",     "&kp K",    "&kp L", "&kp SEMI",   "&kp SQT",
+       "&kp LSHFT",               "&kp Z",           "&kp X",    "&kp C",     "&kp V",  "&kp B",                                     "&kp HOME",                     "&kp PG_UP",                                    "&kp N",  "&kp M", "&kp COMMA",  "&kp DOT", "&kp FSLH", "&kp RSHFT",
+           "&mo 2", "&macro_mac_profile1", "&kp PRINTSCREEN", "&kp LEFT", "&kp RIGHT",                      "&kp BSPC",   "&kp DEL",  "&kp END",                     "&kp PG_DN", "&kp ENTER",         "&kp SPACE",           "&kp UP",  "&kp DOWN", "&kp LBKT", "&kp RBKT",     "&mo 2"
     ], [
-      "&kp EQUAL ", "&kp N1    ", "&kp N2   ", "&kp N3   ", "&kp N4     ", "&kp N5 ", "&trans                                                                          ", "&mo 3  ", "&kp N6 ", "&kp KP_NUM ", "&kp KP_EQUAL ", "&kp KP_DIVIDE ", "&kp KP_MULTIPLY ", "&kp MINUS",
-      "&kp TAB   ", "&kp Q     ", "&kp W    ", "&kp E    ", "&kp R      ", "&kp T  ", "&none                                                                           ", "&none  ", "&kp Y  ", "&kp KP_N7  ", "&kp KP_N8    ", "&kp KP_N9     ", "&kp KP_MINUS    ", "&kp BSLH",
-      "&kp ESC   ", "&kp A     ", "&kp S    ", "&kp D    ", "&kp F      ", "&kp G  ", "&none          ", "&kp LCTRL ", "&kp LALT ", "&kp LGUI  ", "&kp RCTRL           ", "&none  ", "&kp H  ", "&kp KP_N4  ", "&kp KP_N5    ", "&kp KP_N6     ", "&kp KP_PLUS     ", "&kp SQT",
-      "&kp LSHFT ", "&kp Z     ", "&kp X    ", "&kp C    ", "&kp V      ", "&kp B        ",                            "&kp HOME ", "&kp PG_UP "                                   , "&kp N  ", "&kp KP_N1  ", "&kp KP_N2    ", "&kp KP_N3     ", "&kp KP_ENTER    ", "&kp RSHFT",
-      "&mo 2     ", "&kp GRAVE ", "&kp CAPS ", "&kp LEFT ", "&kp RIGHT                   ", "&kp BSPC ", "&kp DEL   ", "&kp END  ", "&kp PG_DN ", "&kp ENTER ", "&kp KP_N0                    ", "&kp UP     ", "&kp DOWN     ", "&kp KP_DOT    ", "&kp RBKT        ", "&mo 2"
+       "&kp EQUAL", "&kp N1", "&kp N2",   "&kp N3",    "&kp N4", "&kp N5",             "&trans",                                                                            "&mo 3", "&kp N6", "&kp KP_NUM", "&kp KP_EQUAL", "&kp KP_DIVIDE", "&kp KP_MULTIPLY", "&kp MINUS",
+         "&kp TAB",  "&kp Q",  "&kp W",    "&kp E",     "&kp R",  "&kp T",              "&none",                                                                            "&none",  "&kp Y",  "&kp KP_N7",    "&kp KP_N8",     "&kp KP_N9",    "&kp KP_MINUS",  "&kp BSLH",
+         "&kp ESC",  "&kp A",  "&kp S",    "&kp D",     "&kp F",  "&kp G",              "&none", "&kp LCTRL", "&kp LALT",                      "&kp LGUI", "&kp RCTRL",     "&none",  "&kp H",  "&kp KP_N4",    "&kp KP_N5",     "&kp KP_N6",     "&kp KP_PLUS",   "&kp SQT",
+       "&kp LSHFT",  "&kp Z",  "&kp X",    "&kp C",     "&kp V",  "&kp B",                                    "&kp HOME",                     "&kp PG_UP",                            "&kp N",  "&kp KP_N1",    "&kp KP_N2",     "&kp KP_N3",    "&kp KP_ENTER", "&kp RSHFT",
+          "&trans", "&trans", "&trans", "&kp LEFT", "&kp RIGHT",                     "&kp BSPC",   "&kp DEL",  "&kp END",                     "&kp PG_DN", "&kp ENTER", "&kp KP_N0",               "&kp UP",     "&kp DOWN",    "&kp KP_DOT",        "&kp RBKT",    "&trans"
     ], [
-      "&kp F1    ", "&kp F2    ", "&kp F3   ", "&kp F4   ", "&kp F5     ", "&kp F6 ", "&tog 1                                                                       ", "&mo 3  ", "&kp F7 ", "&kp F8 ", "&kp F9    ", "&kp F10  ", "&kp F11  ", "&kp F12",
-      "&trans    ", "&trans    ", "&trans   ", "&trans   ", "&trans     ", "&trans ", "&none                                                                        ", "&none  ", "&trans ", "&trans ", "&trans    ", "&trans   ", "&trans   ", "&trans ",
-      "&trans    ", "&trans    ", "&trans   ", "&trans   ", "&trans     ", "&trans ", "&none          ", "&trans    ", "&trans   ", "&trans    ", "&trans           ", "&none  ", "&trans ", "&trans ", "&trans    ", "&trans   ", "&trans   ", "&trans ",
-      "&trans    ", "&trans    ", "&trans   ", "&trans   ", "&trans     ", "&trans       ",                            "&trans   ", "&trans    ",                                 "&trans ", "&trans ", "&trans    ", "&trans   ", "&trans   ", "&trans ",
-      "&trans    ", "&trans    ", "&trans   ", "&trans   ", "&trans                      ", "&trans   ", "&trans    ", "&trans   ", "&trans    ", "&trans    ", "&trans                   ", "&trans ", "&trans    ", "&trans   ", "&trans   ", "&trans"
+       "&kp F1", "&kp F2", "&kp F3", "&kp F4", "&kp F5", "&kp F6",           "&tog 1",                                                              "&mo 3", "&kp F7", "&kp F8", "&kp F9", "&kp F10", "&kp F11", "&kp F12",
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",            "&none",                                                              "&none", "&trans", "&trans", "&trans",  "&trans",  "&trans",  "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",            "&none", "&trans", "&trans",                     "&trans", "&trans",  "&none", "&trans", "&trans", "&trans",  "&trans",  "&trans",  "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",                               "&trans",                     "&trans",                     "&trans", "&trans", "&trans",  "&trans",  "&trans",  "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans",                     "&trans", "&trans", "&trans",                     "&trans", "&trans", "&trans",           "&trans", "&trans",  "&trans",  "&trans",  "&trans"
     ], [
-      "&none ", "&bt BT_SEL 0 ", "&bt BT_SEL 1 ", "&bt BT_SEL 2 ", "&bt BT_SEL 3 ", "&bt BT_SEL 4 ", "&none                                                                                        ", "&trans        ", "&none        ", "&none        ", "&none        ", "&none        ", "&none        ", "&none",
-      "&none ", "&none        ", "&none        ", "&none        ", "&none        ", "&none        ", "&bootloader                                                                                  ", "&bootloader   ", "&none        ", "&none        ", "&none        ", "&none        ", "&none        ", "&none",
-      "&none ", "&none        ", "&none        ", "&none        ", "&none        ", "&none        ", "&none                ", "&none", "&none ", "&bt BT_CLR ", "&none ",                             "&rgb_ug RGB_MEFS_CMD 5 ", "&none        ", "&none        ", "&none        ", "&none        ", "&none        ", "&none",
-      "&none ", "&none        ", "&none        ", "&none        ", "&macro_ver   ", "&none                                          ", "&none ", "&none ",                                                              "&none        ", "&none        ", "&none        ", "&none        ", "&none        ", "&none",
-      "&none ", "&none        ", "&none        ", "&none        ", "&none                                         ", "&none", "&none", "&none ", "&none      ", "&bl BL_TOG ", "&rgb_ug RGB_TOG                                ", "&bl BL_INC   ", "&bl BL_DEC   ", "&none        ", "&none        ", "&none"
+        "&none",        "&bt BT_SEL 0",        "&bt BT_SEL 1", "&bt BT_SEL 2", "&bt BT_SEL 3", "&bt BT_SEL 4",                 "&none",                                                                                     "&trans",  "&none", "&out OUT_USB", "&out OUT_BLE",  "&none",  "&none",                 "&none",
+        "&none",               "&none",               "&none",        "&none",        "&none",        "&none",           "&bootloader",                                                                                "&bootloader",  "&none",        "&none",        "&none",  "&none",  "&none",                 "&none",
+        "&none",               "&none",               "&none",        "&none",        "&none",        "&none",                 "&none",  "&none",  "&none",                     "&bt BT_CLR",      "&none", "&rgb_ug RGB_MEFS_CMD 5",  "&none",        "&none",        "&none",  "&none",  "&none",                 "&none",
+        "&none",               "&none",               "&none",        "&none",        "&none",        "&none",                                     "&none",                          "&none",                                          "&none",        "&none",        "&none",  "&none",  "&none",                 "&none",
+        "&none", "&macro_mac_profile1", "&macro_mac_profile3",        "&none",        "&none",                                 "&none",  "&none",  "&none",                          "&none", "&bl BL_TOG",        "&rgb_ug RGB_TOG",             "&bl BL_INC",   "&bl BL_DEC",  "&none",  "&none", "&macro_linux_profile0"
+    ], [
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",           "&trans",                                                                                   "&mo 3", "&trans", "&trans", "&trans", "&trans", "&trans",                "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",           "&trans",                                                                                  "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",                "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",           "&trans", "&kp LEFT_COMMAND", "&kp LEFT_ALT",                     "&kp LGUI", "&kp RCTRL", "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",                "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans", "&trans",                                               "&trans",                       "&trans",                        "&trans", "&trans", "&trans", "&trans", "&trans",                "&trans",
+       "&trans", "&trans", "&trans", "&trans", "&trans",                     "&trans",           "&trans",       "&trans",                       "&trans",    "&trans", "&trans",           "&trans", "&trans", "&trans", "&trans", "&macro_linux_profile0"
     ]
   ]
-}
+}
\ No newline at end of file
diff --git config/macros.dtsi config/macros.dtsi
index 953577a..a0e3bfa 100644
--- config/macros.dtsi
+++ config/macros.dtsi
@@ -1,36 +1,78 @@
-  macro_quotes: macro_quotes {
-    compatible = "zmk,behavior-macro";
-    label = "macro_quotes";
-    #binding-cells = <0>;
-    bindings = <&kp SQT>, <&kp SQT>, <&kp LEFT>;
-  };
-  macro_dquotes: macro_dquotes {
-    compatible = "zmk,behavior-macro";
-    label = "macro_dquotes";
-    #binding-cells = <0>;
-    bindings = <&kp DQT>, <&kp DQT>, <&kp LEFT>;
-  };
-  macro_braces: macro_braces {
-    compatible = "zmk,behavior-macro";
-    label = "macro_braces";
-    #binding-cells = <0>;
-    bindings = <&kp LBRC>, <&kp RBRC>, <&kp LEFT>;
-  };
-  macro_parens: macro_parens {
-    compatible = "zmk,behavior-macro";
-    label = "macro_parens";
-    #binding-cells = <0>;
-    bindings = <&kp LPAR>, <&kp RPAR>, <&kp LEFT>;
-  };
-  macro_brackets: macro_brackets {
-    compatible = "zmk,behavior-macro";
-    label = "macro_brackets";
-    #binding-cells = <0>;
-    bindings = <&kp LBKT>, <&kp RBKT>, <&kp LEFT>;
-  };
-  macro_kinesis: macro_kinesis {
-    compatible = "zmk,behavior-macro";
-    label = "macro_kinesis";
-    #binding-cells = <0>;
-    bindings = <&kp LS(K)>, <&kp I>, <&kp N>, <&kp E>, <&kp S>, <&kp I>, <&kp S>;
-  };
+macro_mac_profile3: macro_mac_profile3{
+compatible = "zmk,behavior-macro";
+label = "macro_mac_profile3";
+#binding-cells = <0>;
+bindings = <&bt BT_SEL 3>, <&tog 4>;
+};
+macro_mac_profile1: macro_mac_profile1{
+compatible = "zmk,behavior-macro";
+label = "macro_mac_profile1";
+#binding-cells = <0>;
+bindings = <&bt BT_SEL 1>, <&tog 4>;
+};
+macro_linux_profile0: macro_linux_profile0{
+compatible = "zmk,behavior-macro";
+label = "macro_linux_profile0";
+#binding-cells = <0>;
+bindings = <&tog 4>, <&bt BT_SEL 0>;
+};
+macro_command_v: macro_command_v{
+compatible = "zmk,behavior-macro";
+label = "macro_command_v";
+#binding-cells = <0>;
+bindings = <&kp LEFT_COMMAND>, <&kp V>;
+};
+macro_command_x: macro_command_x{
+compatible = "zmk,behavior-macro";
+label = "macro_command_x";
+#binding-cells = <0>;
+bindings = <&kp LEFT_COMMAND>, <&kp X>;
+};
+macro_command_c: macro_command_c{
+compatible = "zmk,behavior-macro";
+label = "macro_command_c";
+#binding-cells = <0>;
+bindings = <&kp LEFT_COMMAND>, <&kp C>;
+};
+macro_command_t: macro_command_t{
+compatible = "zmk,behavior-macro";
+label = "macro_command_t";
+#binding-cells = <0>;
+bindings = <&kp LEFT_COMMAND>, <&kp T>;
+};
+macro_quotes: macro_quotes{
+compatible = "zmk,behavior-macro";
+label = "macro_quotes";
+#binding-cells = <0>;
+bindings = <&kp SQT>, <&kp SQT>, <&kp LEFT>;
+};
+macro_dquotes: macro_dquotes{
+compatible = "zmk,behavior-macro";
+label = "macro_dquotes";
+#binding-cells = <0>;
+bindings = <&kp DQT>, <&kp DQT>, <&kp LEFT>;
+};
+macro_braces: macro_braces{
+compatible = "zmk,behavior-macro";
+label = "macro_braces";
+#binding-cells = <0>;
+bindings = <&kp LBRC>, <&kp RBRC>, <&kp LEFT>;
+};
+macro_parens: macro_parens{
+compatible = "zmk,behavior-macro";
+label = "macro_parens";
+#binding-cells = <0>;
+bindings = <&kp LPAR>, <&kp RPAR>, <&kp LEFT>;
+};
+macro_brackets: macro_brackets{
+compatible = "zmk,behavior-macro";
+label = "macro_brackets";
+#binding-cells = <0>;
+bindings = <&kp LBKT>, <&kp RBKT>, <&kp LEFT>;
+};
+macro_kinesis: macro_kinesis{
+compatible = "zmk,behavior-macro";
+label = "macro_kinesis";
+#binding-cells = <0>;
+bindings = <&kp LS(K)>, <&kp I>, <&kp N>, <&kp E>, <&kp S>, <&kp I>, <&kp S>;
+};
\ No newline at end of file
diff --git config/west.yml config/west.yml
index 9f3a48d..137ea40 100644
--- config/west.yml
+++ config/west.yml
@@ -7,7 +7,8 @@ manifest:
   projects:
     - name: zmk
       remote: refil
-      revision: adv360-z3.5
+      revision: adv360-z3.2
+      #revision: adv360-z3.5
       import: app/west.yml
   self:
     path: config
